torch/testing/_internal/common_fsdp.py

```diff
@@ -517,158 +836,141 @@ def _train_for_several_steps(
+            ref_init_fn (Optional[Callable]): A callable to invoke that wraps a
+                non-wrapped model to construct the reference model. If
+                ``None``, then the callable defaults to the DDP constructor.
+        """
+        assert fsdp_init_mode != FSDPInitMode.NO_FSDP, "Expects an FSDP init mode that wraps with FSDP"
+        if init_kwargs is None:
+            init_kwargs = {}
+        lr = 1e-2
+        rank = self.process_group.rank()
+        # Establish reference behavior with DDP
+        model = model_class.init(
+            self.process_group,
+            FSDPInitMode.NO_FSDP,
+            CUDAInitMode.CUDA_BEFORE,
+            deterministic=True,
+            **init_kwargs,
+        )
+        if ref_init_fn is None:
+            ref_model = DDP(model, device_ids=[rank], output_device=rank)
         else:
-            model = ref_ddp_fn(model)
-
-        # DDP training
+            ref_model = ref_init_fn(model)
         ref_loss = self._train_for_several_steps(
-            model, num_steps, autocast=mixed_precision is not None, lr=lr,
-            fsdp_cpu_offload=cpu_offload, mixed_precision=mixed_precision,
-            enable_sharded_grad_scaler=enable_sharded_grad_scaler,
+            ref_model,
+            num_iters,
+            autocast=mixed_precision is not None,
+            lr=lr,
+            fsdp_cpu_offload=cpu_offload,
+            mixed_precision=mixed_precision,
+            norm_type=norm_type,
+            use_sharded_grad_scaler=use_sharded_grad_scaler,
+        )
+        ddp_params = list(ref_model.parameters())
+        # Check against FSDP behavior
+        fsdp_kwargs.update(
+            {
+                "cpu_offload": cpu_offload,
+                "backward_prefetch": backward_prefetch,
+                "forward_prefetch": forward_prefetch,
+                "sharding_strategy": sharding_strategy,
+                "mixed_precision": mixed_precision,
+            }
         )
-        ref_full_params = list(model.parameters())
-
-        # Confirm we get the same behavior using FullyShardedDataParallel.
         try:
-            model = model_init_fn(
-                group=group,
-                wrap_fsdp=True,
-                fsdp_init_mode=fsdp_init_mode,
-                cpu_offload=cpu_offload,
-                backward_prefetch=backward_prefetch,
-                forward_prefetch=forward_prefetch,
-                sharding_strategy=sharding_strategy,
-                mixed_precision=mixed_precision,
+            fsdp_model = model_class.init(
+                self.process_group,
+                fsdp_init_mode,
+                cuda_init_mode,
+                fsdp_kwargs,
+                deterministic=True,
+                **init_kwargs,
             )
         except Exception as e:
-            raise ValueError(f"model_Init_fn {model_init_fn} got error {str(e)}")
-
-        cpu_offload = cpu_offload or CPUOffload()  # disabled if not specified.
```

Comment:
I removed this to let the FSDP constructor take care of the default argument initialization instead of doing it in this method externally.